lib/pack/localpack/packageserver.go

```diff
@@ -327,22 +327,14 @@ func (p *PackageServer) UpsertRepository(repository string, expires time.Time) e
 		return trace.BadParameter(
 			"expected a valid domain name, got '%v'", repository)
 	}
-	_, err := p.backend.GetRepository(repository)
-	if err == nil {
-		return nil
-	}
-	if !trace.IsNotFound(err) {
-		return trace.Wrap(err)
-	}
 	repo := storage.NewRepository(repository)
 	if !expires.IsZero() {
 		repo.SetExpiry(expires)
 	}
-	_, err = p.backend.CreateRepository(repo)
-	if err != nil {
+	_, err := p.backend.CreateRepository(repo)
```

Comment:
Bit weird that the function is called "UpsertRepository" yet it only creates it. Prob. not a big deal since it's been like this forever, only just noticing.

Reply:
This should propagate to the storage interface which should offer `UpsertRepository` in the same vein. Not going to change that though in this PR.